tpm2/update.go

```diff
@@ -201,12 +203,21 @@ func updateKeyPCRProtectionPolicyCommon(tpm *tpm2.TPMContext, keys []*SealedKeyO
 		// and dynamic authorization policy signing key, so this is the only check required to determine
 		// if 2 keys are related.
 		if !bytes.Equal(k.data.Public().AuthPolicy, primaryKey.data.Public().AuthPolicy) {
-			return InvalidKeyDataError{fmt.Sprintf("key data at index %d is not related to the primary key data", i)}
+			return InvalidKeyDataError{fmt.Sprintf("key data at index %d is not related to the primary key data", i+1)}
 		}
 
 		k.data.Policy().SetPCRPolicyFrom(primaryKey.data.Policy())
 	}
 
+	for _, k := range keys {
+		if k.k == nil {
+			continue
+		}
+		if err := k.k.MarshalAndUpdatePlatformHandle(k); err != nil {
+			return xerrors.Errorf("cannot update TPM platform handle on KeyData: %w", err)
+		}
```

Comment:
because of the k.k accesses it feels like this bit of code should be some sort of helper method on SealedKeyObject

Reply:
This is fixed by PR #250 now.